_posts/2018-01-04-notification-channels-android-o.md

```diff
@@ -0,0 +1,129 @@
+
+So how then does one add channels to notifications? At the simplest level, one needs to create a channel with a unique name and ID and a level of importance. The uniqueness of the name and ID only refers to your app package. The notification manager then does the creation of the channel with whatever settings you specify. Typically the creation code looks like:
+
+```kotlin
+val privateMessagesChannel = NotificationChannel(
+                    PRIVATE_MESSAGES_CHANNEL_ID,
+                    context.getString(R.string.pm_channel_name),
+                    NotificationManager.IMPORTANCE_DEFAULT)
+
+notificationManager.createNotificationChannel(privateMessagesChannel)
+```
+It's worthy of note that you can also create channel groups, and this is useful in scenarious where you have multiple accounts or profiles, so you can group the notification channels per account/profile.
+
+You can specify further customizations for the channel you have created. For example, you can specify the light colors, vibration pattern etc for the channel as shown below:
+
+```kotlin
+with(privateMessagesChannel) {
+    lightColor = Color.RED
+    enableVibration(true)
+    vibrationPattern = longArrayOf(100, 200, 300, 400)
+}
+```
+
+### Sending notifications for a notification channel
+
+Now that we've created the channel, next step is to actually specify the channel whenever we want to show a notification and that is done when creating a new notification.
+Specifying a channel for the notification looks like:
+
+```kotlin
+val notification = NotificationCompat.Builder(context)
+                .setSmallIcon(R.drawable.ic_notification_small)
+                .setContentTitle(context.getString(R.string.notif_pm_title))
+                .setContentText(message)
+                /**
+                 * further notification customizations
+                 */
+                .setChannelId(PRIVATE_MESSAGES_CHANNEL_ID)
+                .build()
+        notificationManager.notify(PM_NOTIFICATION_ID, notification)
+```
+The part of the code above that sets the channel is `.setChannelId()`. Without this line, if you're targeting Android O and above, your notification will not be shown to the user.
+
+### Updating/reading notification channels settings
+
+After creating notification channels, the ability to control is left to the user. They can turn off notifications, change priority, lights & vibration settings etc for any channel they desire. However, it is possible to programmatically read the current channel settings. You may need this, so that you can present the status of the notification channel to your user - for example, if the user has turned off notification for a certain group
```

Comment:
_lights & vibration settings etc for any channel_

_lights **and** vibration settings **etc.** for any channel_

-----------------------------------------

_You may need this, so that you can present the status of the notification channel to your user - for example, if the user has turned off notification_

_You may need this so that you can present the status of the notification channel to your **user. For** example, if the user has turned off notification_

Reply:
🙈 🔒